### PR TITLE
Fix way master and port are fetched from URI

### PR DESCRIPTION
**Pull request by rabbitonweb**
There was a bug in how sparkUri was created. This PR fixes this bug.

### CODE REVIEW

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala [hunk 1]
```diff
@@ -66,8 +68,12 @@ package object hdfs {
 
     val uriOrErr: DefinitionError \/ Uri = Uri.fromString(connUri.value).leftMap((pf: ParseFailure) => liftErr(pf.toString))
 
-    val sparkConfOrErr: DefinitionError \/ SparkConf = uriOrErr.map(uri => 
-      (master(uri) *> appName *>
+    val sparkConfOrErr: DefinitionError \/ SparkConf = for {
+      uri <- uriOrErr
+      host <- uri.host.fold(NonEmptyList("host not provided").left[EnvironmentError].left[Uri.Host])(_.right[DefinitionError])
+      port <- uri.port.fold(NonEmptyList("port not provided").left[EnvironmentError].left[Int])(_.right[DefinitionError])
+    } yield {
```
Comment (djspiewak) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: The above is rather hard to follow.  Are there nested eithers here?

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala [hunk 2]
```diff
@@ -57,7 +58,8 @@ package object hdfs {
 
     def liftErr(msg: String): DefinitionError = NonEmptyList(msg).left[EnvironmentError]
 
-    def master(uri: Uri): State[SparkConf, Unit] = State.modify(_.setMaster(s"spark://${uri.host}:${uri.port}"))
+    def master(host: String, port: Int): State[SparkConf, Unit] =
```
Comment (djspiewak) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: I'm not clear on exactly what the bug was here?

Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: Bug was that `Uri.host` returns `Option[Uri.Host]` and `Uri.port` returns `Option[Int]`

I thought that the returned `String`s

Method `master` was expecting two `String`s, Option was implicitly casted to `String` (because Java ¯\_(ツ)_/¯)

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala (hunk 1 above)
Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: As above, `uri.host` & `uri.port` return `Option` thus I'm folding them to `DefinitionError` if `None`

Comment (rabbitonweb) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: `type DefinitionError  = NonEmptyList[String] \/ EnvironmentError`

Comment (djspiewak) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: Ah, I see.  Well, just as a mental TODO, I think it would probably be better if `DefinitionError` were a proper ADT, rather than encoding it via a coproduct.  Generally better API and error messages that way.  Doesn't hold up the PR though.

File: sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala (hunk 2 above)
Comment (djspiewak) [inline sparkcore/src/main/scala/quasar/physical/sparkcore/fs/hdfs/package.scala]: In general, I don't like replacing semantic types (e.g. `Uri`) with non-semantic, non-newtyped "equivalents".  In this case though, the difference is pretty minor, and `Uri` was only being used to carry around those two values.  👍 from me.
